File: C language/DS/LinkedList/ReverseGetNode1.c
#include<Stdio.h>
#include<stdlib.h>

int num;

struct node{
    int data;
    struct node *next;
};

struct node* AddLast(struct node * head, int value){
    struct node * temp = malloc(sizeof(struct node));
    temp -> data = value;
    temp -> next = NULL;
    if(head == NULL){
        return temp;
    }
    else{
        struct node *traverse = head;
        while(traverse -> next != NULL){
            traverse = traverse->next;
        }
        traverse -> next = temp;
        return(head);
    }
}


void printList(struct node *head){
    struct node *temp = head;
    while(temp != NULL){
        printf("%d ", temp -> data);
        temp = temp -> next;
    }
    printf("\n");
}

void Deallocate(struct node *head){
    struct node *temp = head;
    while(temp != NULL){
        head = temp -> next;
        free(temp);
        temp = head;
    }
}

void printReverse(struct node *head) {
    if(head == NULL){
        return;
    }
    else{
        printReverse(head -> next);
        printf("%d ",head -> data);
    }
}

void revGetNode(struct node *head, int index){
    if(head == NULL){
            return;
        }
        else{
            revGetNode(head -> next, index);
            if(num == index){
                printf("%d\n",head -> data);
            }
            num++;
        }
}

int main(){
    int choice, value, index;
    struct node *head = NULL;

    printf("1. Add 2. Print 3. exit\n");

    do{
        printf("Enter choice : ");
        scanf("%d", &choice);
        if(choice == 1){
            printf("Enter value : ");
            scanf("%d", &value);
            head = AddLast(head, value);
        }
        else if(choice == 2){
            printList(head);
        }
        else if(choice == 3){
            break;
        }
    }while(1);

    do{
        num = 1;
        printf("Enter the index : ");
        scanf("%d", &index);
        if(index == -1){
            break;
        }
        revGetNode(head, index);
    }while(1);
    
    Deallocate(head);
}

File: Java/DS/LinkedList/ReverseGetNode1.java
import java.util.Scanner;

class Node{
    int data;
    Node next;
}

class ReverseGetNode1{
    static int num;
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        Node head = null;
        System.out.println("1. Add 2. Print 3. Exit");
        do{
            System.out.print("Enter choice : ");
            int choice = sc.nextInt();
            if(choice == 1){
                System.out.print("Enter value : ");
                int value = sc.nextInt();
                head = AddLastEle(head, value);
            }
            else if(choice == 2){
                printList(head);
            }
            else if(choice == 3){
                break;
            }
        }while(true);
        do{
            System.out.print("Enter the index : ");
            num = 1;
            int index = sc.nextInt();
            if(index == -1){
                break;
            }
            revGetNode(head, index);
        }while(true);
        sc.close();
    }

    public static int getCount(Node head){
        Node temp = head;
        int count = 0;
        while(temp != null){
            count++; 
            temp = temp.next;
        }
        return count;
    }

    public static void revGetNode(Node head, int index) {
        if(head == null){
            return;
        }
        else{
            revGetNode(head.next, index);
            if(num == index){
                System.out.println(head.data);
            }
            num++;
        }
    }

    public static void printList(Node head){
        Node temp = head;
        while(temp != null){
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
        System.out.println();
    }

    public static Node AddLastEle(Node head, int value){
        Node temp = new Node();
        temp.data = value;
        temp.next = null;
        if(head == null){
            return temp;
        }
        Node traverse = head;
        while(traverse.next != null){
            traverse = traverse.next;
        }
        traverse.next = temp;
        return head;
    }
}